pkg/state/service.go

```diff
@@ -33,6 +33,30 @@ var serviceTableSchema = &memdb.TableSchema{
 // ServicesCollection stores and indexes Kong Services.
 type ServicesCollection collection
 
+func (k *ServicesCollection) AddIgnoringDuplicates(service Service) error {
+	// Detect duplicates
+	if !utils.Empty(service.ID) {
```

Comment:
Is it possible for this to be called with `service`, `service.ID` or `service.Name` being `nil`? If so this will `panic`.

Reply:
`Add()` has a check, so it won't panic. I'll add a test case to demonstrate